cyw920829m2evk_02: failing to build test tests/drivers/flash/common/drivers.flash.common.default

**Describe the bug**

Running twister with command:
`twister -p cyw920829m2evk_02 -s tests/drivers/flash/common/drivers.flash.common.default`
fails with compilation errors on implicit declaration of `cyhal_flash_*` functions,
There is also Null-pointer exception misconfiguration issue appearing when building the test.

**To Reproduce**
Steps to reproduce the behavior:
1. go to Zephyr root, switch to main branch and bring latest version
2. . ./twister-env.sh
3. west update
4. twister -p cyw920829m2evk_02 -s tests/drivers/flash/common/drivers.flash.common.default
5. see error as in logs below

**Expected behavior**
Not failing build.

**Impact**
Impacts PRs that modify anything around Cypress flash driver, CI blocker.

**Logs and console output**
```
arch/arm/core/mpu/arm_mpu.c:475:9: note: '#pragma message: Null-Pointer exception detection cannot be configured on un-mapped flash areas'
  475 | #pragma message "Null-Pointer exception detection cannot be configured on un-mapped flash areas"
(...)
CMakeFiles/drivers__flash.dir/flash_ifx_cat1.c.obj -c /home/byte/other_src/zephyrproject/zephyr/drivers/flash/flash_ifx_cat1.c
In file included from /home/byte/other_src/zephyrproject/zephyr/drivers/flash/flash_ifx_cat1.c:18:
/home/byte/other_src/zephyrproject/zephyr/drivers/flash/flash_ifx_cat1.c: In function 'ifx_cat1_flash_read':
/home/byte/other_src/zephyrproject/modules/hal/infineon/mtb-hal-cat1/include/cyhal_flash.h:118:49: error: implicit declaration of function 'cyhal_nvm_read'; did you mean 'cyhal_flash_read'? [-Werror=implicit-function-declaration]
  118 | #define cyhal_flash_read                        cyhal_nvm_read
      |                                                 ^~~~~~~~~~~~~~
/home/byte/other_src/zephyrproject/modules/hal/infineon/mtb-hal-cat1/include/cyhal_flash.h:118:49: note: in definition of macro 'cyhal_flash_read'
  118 | #define cyhal_flash_read                        cyhal_nvm_read
      |                                                 ^~~~~~~~~~~~~~
/home/byte/other_src/zephyrproject/zephyr/drivers/flash/flash_ifx_cat1.c: In function 'ifx_cat1_flash_write':
/home/byte/other_src/zephyrproject/modules/hal/infineon/mtb-hal-cat1/include/cyhal_flash.h:148:49: error: implicit declaration of function 'cyhal_nvm_write'; did you mean 'cyhal_flash_write'? [-Werror=implicit-function-declaration]
  148 | #define cyhal_flash_write                       cyhal_nvm_write
      |                                                 ^~~~~~~~~~~~~~~
/home/byte/other_src/zephyrproject/modules/hal/infineon/mtb-hal-cat1/include/cyhal_flash.h:148:49: note: in definition of macro 'cyhal_flash_write'
  148 | #define cyhal_flash_write                       cyhal_nvm_write
      |                                                 ^~~~~~~~~~~~~~~
/home/byte/other_src/zephyrproject/zephyr/drivers/flash/flash_ifx_cat1.c: In function 'ifx_cat1_flash_erase':
/home/byte/other_src/zephyrproject/modules/hal/infineon/mtb-hal-cat1/include/cyhal_flash.h:131:49: error: implicit declaration of function 'cyhal_nvm_erase'; did you mean 'cyhal_flash_erase'? [-Werror=implicit-function-declaration]
  131 | #define cyhal_flash_erase                       cyhal_nvm_erase
      |                                                 ^~~~~~~~~~~~~~~
/home/byte/other_src/zephyrproject/modules/hal/infineon/mtb-hal-cat1/include/cyhal_flash.h:131:49: note: in definition of macro 'cyhal_flash_erase'
  131 | #define cyhal_flash_erase                       cyhal_nvm_erase
      |                                                 ^~~~~~~~~~~~~~~

```
**Environment (please complete the following information):**
 - OS:Debian 12
 - Toolchain: Zephyr SDK 0.16.1)
 - Zephyr Commit 3ea5f799dd7978b7baf2edd7d2458e0c36e431fc

--- a/boards/infineon/cyw920829m2evk_02/cyw920829m2evk_02.dts
+++ b/boards/infineon/cyw920829m2evk_02/cyw920829m2evk_02.dts
@@ -71,8 +71,7 @@ uart2: &scb2 {
 	status = "okay";
 };
 
-&flash_controller {
-
+/ {
 	flash0: flash@60000000 {
 		compatible = "soc-nv-flash";
 		reg = <0x60000000 DT_SIZE_K(512)>;

--- a/dts/arm/infineon/cat1b/cyw20829/cyw20829.dtsi
+++ b/dts/arm/infineon/cat1b/cyw20829/cyw20829.dtsi
@@ -19,13 +19,6 @@
 		};
 	};
 
-	flash_controller: flash_controller@40250000 {
-		compatible = "infineon,cat1-flash-controller";
-		reg = <0x40250000 0x10000>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-	};
-
 	sram0: memory@20000000 {
 		compatible = "mmio-sram";
 		reg = <0x20000000 0x3DC00>;

